binding/client.go

```diff
@@ -765,3 +783,10 @@ func (f *Field) disposition() (d string) {
 	d = fmt.Sprintf(`form-data; name="%s"; filename="%s"`, f.Name, pathlib.Base(f.Path))
 	return
 }
+
+func (r *Client) refreshToken(status int, path string) (refresh bool) {
```

Comment:
I think the refreshToken() should only return an error when it's not a RestError because:
1. send() should return the original 401 response when refresh fails.
2.  The caller methods (in the client) is responsible for interpreting response codes.

Something like:
```
// refreshToken refreshes the token.
func (r *Client) refreshToken(request *http.Request) (refreshed bool, err error) {
    if strings.HasSuffix(request.URL.Path, api.AuthRefreshRoot) {
        return
    }
    login := &api.Login{Refresh: r.token}
    err = r.Post(
        api.AuthRefreshRoot,
        login)
    if err == nil {
        r.token = login.Token
        refreshed = true
        return
    }
    if errors.Is(err, &RestError{}) {
        err = nil
    }
    return
}
```